src/components/MoodSelector/MoodSelector.tsx

```diff
@@ -0,0 +1,193 @@
+import { useLayoutEffect, useRef } from 'react'
+import Div100vh from 'react-div-100vh'
+import styled from 'styled-components/macro'
+import Button from '../common/Button/Button'
+import MoodItem from './MoodItem'
+import CloseIcon from '@material-ui/icons/Close'
+import { Mood } from '../../interfaces/SpotifyExtensions'
+
+interface Props {
+  moods: Mood[]
+  preselectedMoodId: number
+  setShowMoodSelector: React.Dispatch<React.SetStateAction<boolean>>
+  setCurrentMood: React.Dispatch<React.SetStateAction<Mood>>
+}
+
+interface Variables {
+  moodList: HTMLUListElement
+  moodItems: Array<HTMLElement>
+  moodItemCenter: number
+  boxCenter: number
+  bottomLine: number
+  topLine: number
+}
+
+function MoodSelector({
+  moods,
+  preselectedMoodId,
+  setShowMoodSelector,
+  setCurrentMood,
+}: Props): JSX.Element {
+  const moodListRef = useRef<HTMLUListElement>(null)
+  const variables = useRef<Variables | null>(null)
+
+  useLayoutEffect(() => {
```

Comment:
Interisting havent seen that in action :)

Reply:
I would recommend to stick with useEffect though. Interesting article about this here:  [Kentcdodds](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect)
Only reason I use useLayoutEffect is: 

> One other situation you might want to use useLayoutEffect instead of useEffect is if you're updating a value (like a ref) and you want to make sure it's up-to-date before any other code runs